packages/picasso-forms/src/utils/auto-save-decorator/auto-save-decorator.ts

```diff
@@ -0,0 +1,68 @@
+import { AnyObject, FormApi } from 'final-form'
+
+type ChangedFields<T extends AnyObject> = Partial<Record<keyof T, boolean>>
+
+export const getChangedFields = <T extends AnyObject>(
+  newValues: T,
+  existingValues?: T,
+  subscribedFields?: (keyof T)[]
+): ChangedFields<T> => {
```

Comment:
```suggestion
) => {
```
I think it's unnecessary; Also we are returning `result as Record<keyof T, boolean>`, which won't have optional keys set :thinking:

Reply:
You are right. It is not necessary but I kept that to provide type safety for the creator function. If not, `changedFields` will become `{}`. I remove the type from `result` variable while returning, that's not necessary 👍